code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm

```diff
@@ -52,12 +52,24 @@
 
 /datum/eldritch_knowledge/rust_regen
 	name = "Leeching Walk"
-	desc = "Passively heals you when you are on rusted tiles."
+	desc = "Passively heals you and provides stun resistance when you are on rusted tiles."
 	gain_text = "The strength was unparalleled, unnatural. The Blacksmith was smiling."
 	cost = 1
 	next_knowledge = list(/datum/eldritch_knowledge/rust_mark,/datum/eldritch_knowledge/armor,/datum/eldritch_knowledge/essence)
 	route = PATH_RUST
 
+/datum/eldritch_knowledge/rust_regen/on_gain(mob/user)
+	. = ..()
+	RegisterSignal(user,COMSIG_MOVABLE_MOVED,.proc/on_move)
+
+/datum/eldritch_knowledge/rust_regen/proc/on_move(mob/mover)
+	SIGNAL_HANDLER
+
+	if(istype(get_turf(mover),/turf/open/floor/plating/rust))
+		ADD_TRAIT(mover,TRAIT_STUNRESISTANCE,type)
```

Comment:
friendly reminder that stun resistance ONLY reduces the knockdown length of STUN batons, and does not reduce the stamina damage from stun batons (so they'll still two-shot staminacrit you), eliminate the knockdown entirely (so you'll still drop whatever you were holding), or give you any form of protection against literally any other sources of knockdowns (including detective batons and telebatons). TRAIT_STUNRESISTANCE really should be renamed to "TRAIT_STUNBATONRESISTANCE"...

Reply:
i know